packages/firestore/src/local/indexeddb_schema.ts

```diff
@@ -55,9 +61,16 @@ export function createOrUpgradeDb(
   }
 
   if (fromVersion < 2 && toVersion >= 2) {
+    p = ensureTargetGlobalExists(txn).next(targetGlobal =>
+      saveTargetCount(txn, targetGlobal)
+    );
+  }
+
+  if (fromVersion < 3 && toVersion >= 3) {
```

Comment:
I think this should probably be:

```
p = p.next(() => {
     createClientMetadataStore(db);
     createTargetChangeStore(db);
});
```

So we do everything in the right order (not that these migrations are order-dependent right now).